theano/scan_module/scan_op.py

```diff
@@ -2198,7 +2198,7 @@ def compute_all_gradients(known_grads):
                 dC_dinps_t[dx + self.n_seqs] += dC_dXtm1
         # Construct scan op
         # Seqs
-        outer_inp_seqs = [x[::-1] for x in inputs[1:1 + self.n_seqs]]
+        outer_inp_seqs = [x[:grad_steps][::-1] for x in inputs[1:1 + self.n_seqs]]
```

Comment:
Instead of doing 2 subtensor and asking Theano optimization to do the clean up, can you combine both? I think could do it:

    x[grad_steps-1::-1]
```

>>> range(10)[4-1::-1]
[3, 2, 1, 0]
>>> range(10)[:4][::-1]
[3, 2, 1, 0]

```